### PR TITLE
Allow loading .ply with loadFileAsync

### PR DESCRIPTION
**Pull request by CedricGuillemet**
Follow up https://forum.babylonjs.com/t/unable-to-load-gaussian-spalt-ply-file/47707

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts [hunk 1]
```diff
@@ -141,13 +287,13 @@ export class GaussianSplattingMesh extends Mesh {
...
     /**
-     * Loads a .splat Gaussian Splatting file asynchronously
+     * Loads a .splat Gaussian or .ply Splatting file asynchronously
      * @param url path to the splat file to load
      * @returns a promise that resolves when the operation is complete
      */
     public loadFileAsync(url: string): Promise<void> {
         return Tools.LoadFileAsync(url, true).then((data) => {
-            this._loadData(data);
+            this._loadData(GaussianSplattingMesh.ConvertPLYToSplat(data));
```
Comment (Popov72) [inline packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:296]: Shouldn't you test the extension and call `ConvertPLYToSplat` only if it is ".ply"?

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts [hunk 2]
```diff
@@ -130,6 +130,152 @@ export class GaussianSplattingMesh extends Mesh {
...
     }
 
+    /**
+     * Code from https://github.com/dylanebert/gsplat.js/blob/main/src/loaders/PLYLoader.ts Under MIT license
+     * Converts a .ply data array buffer to splat
+     * if data array buffer is not ply, returns the original buffer
+     * @param data the .ply data to load
+     * @returns the loaded splat buffer
+     */
+    public static ConvertPLYToSplat(data: any): ArrayBuffer {
```
Comment (Popov72) [inline packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:140]: `ArrayBuffer` would be better than `any`.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14761/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14761/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14761/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14761/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14761/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14761/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14761/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14761/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Experimental playwright test reporter:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14761/merge/testResults/webgl2playwright/index.html

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts (hunk 1 above)
Comment (CedricGuillemet) [reply packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts:296]: this was my 1st draft but nothing guarantees there is a .ply extension anywhere in the url (I believe luma SDK urls don't contain one). Moreover, there is a header check that looks more robust to me.

File: packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts (hunk 2 above)
Comment (CedricGuillemet) [reply packages/dev/core/src/Meshes/GaussianSplatting/gaussianSplattingMesh.ts]: Fixed.

**Review by RaananW — APPROVED** [review]

**Review by Popov72 — APPROVED** [review]

**Pull request merged**